packages/constellation/src/components/Compositions/Input/Input.tsx

```diff
@@ -0,0 +1,84 @@
+import classnames from 'classnames'
+import React, { forwardRef, useState } from 'react'
+
+import { IconVisibilityOff, IconVisibilityOn } from '../..'
+import { InputProps } from './Input.types'
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+  (
+    {
+      disabled,
+      helperText = '',
+      id,
+      label = '',
+      placeholder,
+      prefix,
+      state,
+      type = 'text',
+      ...args
+    },
+    ref,
+  ) => {
+    const [showPassword, setShowPassword] = useState(false)
+    const password = type === 'password'
+
+    return (
+      <div className='constellation'>
```

Comment:
random question. why do we need this?

Reply:
Tailwind uses a lot of global styles that could cause styling conflicts when put into an already established consuming app but all classNames in constellation include the constellation selector eg: 
```
.constellation .rounded-lg {
  border-radius: 0.5rem;
}
```
Using this method only component that are descendants of or contain the `constellation` className will be affected by out styles.